### PR TITLE
protocol parser: rdp

### PR DESCRIPTION
**Pull request by lmzke**
Continued in #4174. Continuation of #4095.

Initial implementation of feature [2314](https://redmine.openinfosecfoundation.org/issues/2314):
1. Add protocol parser for RDP
2. Add transactions for RDP negotiation
3. Add eve logging of transactions

In a separate commit, Travis CI rustc minimum bumped to v1.34.0 (Rust 2018), with 2018 needed by dependencies.  Latest known-good version bumped to v1.37.0 (2019-08-15), which is the current rustc stable.

@jasonish mentioned in #4088 that the master/5.x branches may not be ready to leave minimum v1.24.1 yet.  This PR may need to block until then.  See also task [2629](https://redmine.openinfosecfoundation.org/issues/2629).

This PR has incorporated all previous reviewer suggestions.  New contributor, all feedback welcome.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made

Sample output, based on [wireshark RDP-002.pcap](https://wiki.wireshark.org/SampleCaptures?action=AttachFile&do=view&target=RDP-002.pcap.gz):

```json
"rdp": {
  "tx_id": 0,
  "event_type": "initial_request",
  "cookie": "A70067"
}

"rdp": {
  "tx_id": 1,
  "event_type": "initial_response"
}

"rdp": {
  "tx_id": 2,
  "event_type": "connect_request",
  "client": {
    "version": "v5",
    "desktop_width": 1152,
    "desktop_height": 864,
    "color_depth": 15,
    "keyboard_layout": "en-US",
    "build": "Windows XP",
    "client_name": "ISD2-KM84178",
    "keyboard_type": "enhanced",
    "function_keys": 12,
    "product_id": 1,
    "capabilities": [
      "support_errinfo_pdf"
    ],
    "id": "55274-OEM-0011903-00107"
  },
  "channels": [
    "rdpdr",
    "cliprdr",
    "rdpsnd"
  ]
}

"rdp": {
  "tx_id": 3,
  "event_type": "connect_response"
}
```

### CODE REVIEW

**Comment by lmzke** [comment]
I set my Travis CI update in this PR to match then-candidate minimum rustc 1.34.0 in task [2629](https://redmine.openinfosecfoundation.org/issues/2629).

Current candidate minimum there is is 1.33.  The dependencies of this PR cleanly build on 1.32.0 and higher, so the RDP parser should integrate nicely afterwards, with either minimum.

File: src/app-layer-parser.c [hunk 1]
```diff
@@ -1508,6 +1509,7 @@ void AppLayerParserRegisterProtocolParsers(void)
     RegisterSNMPParsers();
     RegisterTemplateRustParsers();
     RegisterTemplateParsers();
+    RegisterRdpRustParsers();
```
Comment (jlucovsky) [inline src/app-layer-parser.c]: Nit -- I'm not sure why this can't be  just `RegisterRdpParsers`?

File: src/app-layer-rdp-rust.c [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/* Copyright (C) 2019 Open Information Security Foundation
```
Comment (jlucovsky) [inline src/app-layer-rdp-rust.c]: Another nit -- Is it necessary to embed `rust` within the filename?

File: src/app-layer-rdp-rust.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/* Copyright (C) 2019 Open Information Security Foundation
```
Comment (jlucovsky) [inline src/app-layer-rdp-rust.h]: Same nit -- why include `rust` in the filename?

File: src/output-json-rdp-rust.c [hunk 4]
```diff
@@ -0,0 +1,180 @@
+/* Copyright (C) 2019 Open Information Security Foundation
```
Comment (jlucovsky) [inline src/output-json-rdp-rust.c]: Same nit -- why `rust` in the filename?

File: src/app-layer-rdp-rust.c (hunk 2 above)
Comment (jasonish) [reply src/app-layer-rdp-rust.c]: No. We should probably get rid of this naming convention. It made sense with DNS when we had a C and a Rust one.  But it doesn't make sense anymore.

**Comment by lmzke** [comment]
Continued in #4174.

**Pull request closed without merging**